docs/docs/client.md

```diff
@@ -423,8 +423,10 @@ httpClient.get[Either[String, String]](endpoint) {
 ```
 
 Your function has to consume the body before the returned `F` exits.
-`Response.body` yields a `EntityBody` which is a type alias for `Stream[F, Byte]`.
-It's this `Stream` that needs to be consumed within your effect `F`.
+`Response.entity` yields an `Entity[F]` that encodes body of the response.
+A response body may be empty, completely loaded into memory, or streamed.
+For more details, see the [entity] page. In the case of a streamed response, 
+it's this `Stream` that needs to be consumed within your effect `F`.
```

Comment:
This is good exposition, but just for discussion: do we have any client that will return a strict entity actually? I'm curious how that works.

Reply:
That straightforwardly comes from types, so it's better to highlight all variants of the `Entity`. Even if the users can't receive the `Strict` right now, they could receive `Empty` at least.

Reply:
> they could receive `Empty` at least.

I am actually not sure if any client does that either, yet. Maybe it would be more straightforward to implement. The tricky thing is the client needs to make some decision about how to treat the entity based on the `Content-Length` I guess—which might not even be the true, correct value.

> That straightforwardly comes from types, so it's better to highlight all variants of the `Entity`.

It's true. I wonder if it is better to talk about this in the context of a request (instead of a response) since that is more practical.

Anyway, don't mean to hold this up, sorry 😅

Reply:
> I am actually not sure if any client does that either, yet. Maybe it would be more straightforward to implement.

But our few `EntityDecoder`s already support the new `Entity` model.  Doesn't that straightforwardly go into the responses on the client side?

Reply:
The `EntityDecoder`s process responses from the client, but it is the responsibility of the client to consume bytes from the socket and package that as either an empty, strict, or streaming entity.

Anyway, actually I was wrong :) it looks like Ember client does return empty bodies.
https://github.com/http4s/http4s/blob/5dae6eefb13c39e6a31665b8f9b21b62d18dcff5/ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala#L419-L420

And even strict bodies!
https://github.com/http4s/http4s/blob/5dae6eefb13c39e6a31665b8f9b21b62d18dcff5/ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala#L557-L559

Reply:
By the way, even [`http4s-armeria` handles `Entity`'s model](https://github.com/http4s/http4s-armeria/blob/series/1.x/client/src/main/scala/org/http4s/armeria/client/ArmeriaClient.scala#L51-L90) (but only for requests at the moment though).